app/models/volunteer.rb

```diff
@@ -33,23 +32,25 @@ class Volunteer < ApplicationRecord
   belongs_to :registrar, optional: true,
     class_name: 'User', foreign_key: 'registrar_id'
 
-  has_many :certificates
+  has_many :certificates, dependent: :destroy
 
-  has_one :contact, as: :contactable
+  has_one :contact, as: :contactable, dependent: :destroy
   accepts_nested_attributes_for :contact
 
   has_many :journals, as: :journalable, dependent: :destroy
   accepts_nested_attributes_for :journals, allow_destroy: true
 
   has_many :assignments, dependent: :destroy
-  has_many :clients, through: :assignments
+  has_many :clients, through: :assignments, dependent: :destroy
```

Comment:
Are you sure you want to delete the clients when a volunteer is deleted? It looks like they could still be associated with other volunteers.